Challenge1/src/agent.spec.ts

```diff
@@ -1,74 +1,108 @@
+  it("should return empty findings", async () => {
+    const mockTxEvent = {
+      from: "0x1234567890123456789012345678901234567890",
+      to: mockFortaRegistryAddress,
+      filterFunction: jest.fn().mockReturnValue([]),
+    } as unknown as TransactionEvent;
+
+    const findings = await handleTransaction(mockTxEvent);
+
+    expect(findings).toHaveLength(0);
+  });
+});
+
+describe("returns empty findings if transaction is not to Forta address", () => {
+  it("should return empty findings", async () => {
+    const mockTxEvent = {
+      from: mockNetherMindAddress,
+      to: "0x1234567890123456789012345678901234567890",
+      filterFunction: jest.fn().mockReturnValue([]),
+    } as unknown as TransactionEvent;
+
+    const findings = await handleTransaction(mockTxEvent);
+
+    expect(findings).toHaveLength(0);
+  });
+});
+
+describe("returns empty findings if neither createAgent nor updateAgent are detected", () => {
+  it("should return empty findings", async () => {
+    const mockTxEvent = {
+      from: mockNetherMindAddress,
+      to: mockFortaRegistryAddress,
+      filterFunction: jest.fn().mockReturnValue([]),
+    } as unknown as TransactionEvent;
+
+    const findings = await handleTransaction(mockTxEvent);
+
+    expect(findings).toHaveLength(0);
+  });
+});
+
+describe("detects bot deployment", () => {
+  it("should return a finding", async () => {
+    const mockTxEvent = {
+      from: mockNetherMindAddress,
+      to: mockFortaRegistryAddress,
+      filterFunction: jest.fn().mockReturnValue([
+        {
+          name: "createAgent",
+          args: { agentId: "123" },
+        },
+      ]),
+    } as unknown as TransactionEvent;
+
+    const findings = await handleTransaction(mockTxEvent);
+
+    expect(findings).toHaveLength(1);
+    expect(findings[0]).toEqual(
+      expect.objectContaining({
+        name: "Nethermind Bot Deployment",
+        description: `Nethermind deployed a new bot with ID: 123`,
+        alertId: "NEW-BOT-DEPLOYED",
+        severity: FindingSeverity.Low,
+        type: FindingType.Info,
+      } as Finding)
+    );
+  });
+});
+
+describe("detects bot update", () => {
+  it("should return a finding", async () => {
+    const mockTxEvent = {
+      from: mockNetherMindAddress,
+      to: mockFortaRegistryAddress,
+      filterFunction: jest.fn().mockReturnValue([
+        {
+          name: "updateAgent",
+          args: { agentId: "456" },
+        },
+      ]),
+    } as unknown as TransactionEvent;
+
+    const findings = await handleTransaction(mockTxEvent);
+
+    expect(findings).toHaveLength(1);
+    expect(findings[0]).toEqual(
+      expect.objectContaining({
+        name: "Nethermind Bot Update",
+        description: `Nethermind updated an existing bot with ID: 456`,
+        alertId: "EXISTING-BOT-UPDATED",
+        severity: FindingSeverity.Low,
+        type: FindingType.Info,
+      } as Finding)
+    );
+  });
+});
```

Comment:
Some additional test cases I would request from you is to make sure the bot doesn't detect bot creation nor bot updates from contracts that aren't the Forta `AgentRegistry`. So similar to your `"returns empty findings if transaction is not to Forta address"` test case, but the transaction would actually include the functions the bot is detecting for, but won't actually alert because the contract is not the correct one.